temporian/beam/operators/drop_index.py

```diff
@@ -0,0 +1,163 @@
+            all_values = []
+            values_are_none = False
+            for item in items:
+                _, (item_timestamps, item_values) = item
+                all_timestamps.append(item_timestamps)
+                all_values.append(item_values)
+                if item_values is None:
+                    values_are_none = True
+
+            aggregated_timestamps = np.concatenate(all_timestamps)
+            sorted_idxs = np.argsort(aggregated_timestamps, kind="mergesort")
+            new_timestamps = aggregated_timestamps[sorted_idxs]
+            if values_are_none:
+                new_features = None
+            else:
+                new_features = np.concatenate(all_values)[sorted_idxs]
+            return new_indexes, (new_timestamps, new_features)
+
+        def src_index_to_feature(
+            any_group: Tuple[BeamIndexKey, Sequence[FeatureItem]],
+            final_nonindex_idx: int,
+        ) -> FeatureItem:
+            """Create a feature with the dropped index."""
+            new_indexes, items = any_group
+            all_timestamps = []
+            all_values = []
+            for item in items:
+                old_index, (item_timestamps, _) = item
+                item_values = np.full(
+                    shape=len(item_timestamps),
+                    fill_value=old_index[final_nonindex_idx],
+                )
+                all_timestamps.append(item_timestamps)
+                all_values.append(item_values)
+            aggregated_timestamps = np.concatenate(all_timestamps)
+            sorted_idxs = np.argsort(aggregated_timestamps, kind="mergesort")
+            new_timestamps = aggregated_timestamps[sorted_idxs]
+            new_features = np.concatenate(all_values)[sorted_idxs]
+            return new_indexes, (new_timestamps, new_features)
+
+        def feature_less_event(
+            any_group: Tuple[BeamIndexKey, Sequence[FeatureItem]],
+        ) -> FeatureItem:
+            """Create an event without feature."""
+            new_indexes, items = any_group
+            all_timestamps = []
+            for item in items:
+                _, (item_timestamps, _) = item
+                all_timestamps.append(item_timestamps)
+            aggregated_timestamps = np.concatenate(all_timestamps)
+            sorted_idxs = np.argsort(aggregated_timestamps, kind="mergesort")
+            new_timestamps = aggregated_timestamps[sorted_idxs]
+            return new_indexes, (new_timestamps, None)
+
+        group_by_new_index = tuple(
+            item
+            | f"Gen new index on feature #{feature_idx} {self.operator}"
+            >> beam.GroupBy(build_new_index)
+            for feature_idx, item in enumerate(input)
+        )
+        assert len(group_by_new_index) >= 1
+
+        has_input_features = len(self.operator.inputs["input"].features) > 0
+
+        if not has_input_features:
+            existing_features_output = ()
+            assert len(group_by_new_index) == 1
```

Comment:
Good question.

In the beam code, the "unit of computation" is a tuple with the timestamps and values for a single feature in a single index value. Those units of computation are grouped into channels according to the feature idx. If an event-set has not features (e.g. has_input_features=False in the code you commented), we create a single channel where the unit of computation contains the timestamps and None (instead of feature values).

This code checks that, if there are no features, there is exactly one channel.